pkg/tools/cloudformationguard/cfg.go

```diff
@@ -0,0 +1,142 @@
+package cloudformationguard
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/soluble-ai/go-jnode"
+	"github.com/soluble-ai/soluble-cli/pkg/client"
+	"github.com/soluble-ai/soluble-cli/pkg/download"
+	"github.com/soluble-ai/soluble-cli/pkg/log"
+	"github.com/soluble-ai/soluble-cli/pkg/tools"
+	"github.com/soluble-ai/soluble-cli/pkg/tools/iacinventory"
+)
+
+const (
+	rulesZip  = "cloudformation-guard-rulesets.zip"
+	rulesPath = "cloudformation-guard"
+)
+
+type Tool struct {
+	Directory string
+	APIClient client.Interface
+
+	rulesPath string
+}
+
+func (t *Tool) Name() string {
+	return "cloudformationguard"
+}
+
+func (t *Tool) Run() (*tools.Result, error) {
+	m := download.NewManager()
+	d, err := m.InstallGithubRelease("aws-cloudformation", "cloudformation-guard", "")
+	if err != nil {
+		return nil, fmt.Errorf("error downloading cloudformation-guard from GitHub: %w", err)
+	}
+	binDir := filepath.Join(d.Dir, "cfn-guard-osx") // TODO: directory varies by name. InstallGithubRelease should extract to consistent dirname.
+	program := binDir + "/cfn-guard"
+
+	if err := t.downloadRules(); err != nil {
+		return nil, fmt.Errorf("error downloading cloudformation-guard rules: %w", err)
+	}
+
+	var ruleFiles []string
+	err = filepath.Walk(t.rulesPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.Mode().IsRegular() {
+			return nil
+		}
+		if filepath.Ext(path) != "ruleset" {
+			return nil
+		}
+		ruleFiles = append(ruleFiles, path)
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error getting rule files: %w", err)
+	}
+
+	// TODO: move this out of inline func
+	type violation struct {
```

Comment:
I left this inlined in case we have some formal convention that I should be following here.